src/main/java/seedu/address/model/order/Date.java

```diff
@@ -0,0 +1,79 @@
+package seedu.address.model.order;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Represents the deliver-by date of a Person's Order in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
+ */
+public class Date {
+    public static final String DATE_FORMATTER = "dd-MM-yyyy";
+    public static final String MESSAGE_CONSTRAINTS = "Dates should be of the format dd-mm-yyyy "
+            + "and adhere to the following constraints:\n"
+            + "1. The day \"dd\" should be a 2-digit number representing a valid day the specified calender month\n"
+            + "2. The month \"mm\" should be a 2-digit number representing a valid month in the calender year\n"
+            + "3. The year \"yyyy\" should be a 4-digit number representing a valid year within the 21st century\n"
+            + "4. The day, month and year are separated by a hyphen '-'.";
+    private static final String REGEX_DAY = "(0[1-9]|[12][0-9]|3[01])";
+    private static final String REGEX_MONTH = "(0[1-9]|1[0-2])";
+    private static final String REGEX_YEAR = "(20[0-9]{2})";
+    public static final String REGEX_DATE = "^" + REGEX_DAY + "-" + REGEX_MONTH + "-" + REGEX_YEAR + "$";
+    public final String value;
+
+    /**
+     * Constructs an {@code Date}.
+     *
+     * @param date A valid date address.
+     */
+    public Date(String date) {
+        requireNonNull(date);
+        checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
+        value = date;
+    }
+
+    /**
+     * Returns if a given string is a valid date.
+     */
+    public static boolean isValidDate(String test) {
+        if (!test.matches(REGEX_DATE)) {
```

Comment:
Wouldn't the `ParserUtil.parseLocalDate` in the method already verify that the string is written in the correct date format? (`ParserUtil.parseLocalDate` throws an exception if the string does not follow the format correctly)

Reply:
This is following the coding style and methods of other Person detail classes. Further, the conditional cases here is to allow for 2 checks for validation of date, where date could be due to poor formatting, or invalid dates like 30 feb, where isValidDate() could then return separate errors for each.